### PR TITLE
Fr1.14

### PR DESCRIPTION
**Pull request by QiYuan7**
Added methods in NewBankClientHandler and NewBank for new user to sign up.

### CODE REVIEW

File: src/newbank/server/NewBank.java [hunk 1]
```diff
@@ -42,7 +42,62 @@ public static NewBank getBank() {
         return bank;
     }
 
+    //TODO: not needed any more
```
Comment (jack27696) [inline src/newbank/server/NewBank.java]: Remove the test Data as this now works with JSON.

File: src/newbank/server/NewBank.java [hunk 2]
```diff
@@ -42,7 +42,62 @@ public static NewBank getBank() {
...
+     */
+    public boolean passWordComplexity(String password){
+        return (password.length()>=8);
+    }
+    /**
+     * To hash the user password for database storage.
+     * @param password
+     * @return
+     */
+    private String hashPassword(String password){
```
Comment (jack27696) [inline src/newbank/server/NewBank.java:89]: Has this been added to be implemented at a later date?

**Review by jack27696 — CHANGES_REQUESTED** [review]
The code looks generally good. Just a few cleanup bits to do. Thanks!

File: src/newbank/server/NewBank.java (hunk 1 above)
Comment (QiYuan7) [reply src/newbank/server/NewBank.java]: Done, all code involves using the test Data now removed.

File: src/newbank/server/NewBank.java (hunk 2 above)
Comment (QiYuan7) [reply src/newbank/server/NewBank.java:89]: Yes, this is to be implemented by other FR(FR1.12); the declaration leaves room for future implement.

**Review by jack27696 — APPROVED** [review]

**Comment by jack27696** [comment]
Happy with changes merging into main

**Pull request merged**